### ISSUE TITLE
[mesheryctl] Migrate 'system config eks' to golang SDK

### ISSUE THREAD

**Issue by hexxdump** [meshery/meshery#2842]
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior
mesheryctl system config eks command invokes the below cli from scripts:
“aws eks update-kubeconfig --region <regionName> --name <clusterName> -kubeconfig <configPath>” CLI.

#### Desired Behavior
Instead of using aws CLI, use golang SDKs provided by AWS to achieve the same result.

#### Environment

- Host OS: 
- Meshery Version: 
- Platform [docker/kubernetes/gke/eks etc]: eks

---

#### Implementation
We should use https://github.com/aws/aws-sdk-go-v2  to get EKS cluster’s kube config file, i.e., using https://github.com/aws/aws-sdk-go-v2/tree/main/service/eks

Background of the issue: [here](https://docs.google.com/document/d/1XfIvMwKGKBS5_ielGWuYFdqJByEcOeloCKpUL9bBxlw/edit#)
#### Mockups

---

#### Contributor Resources

- [mesheryctl Contributing Guide](https://github.com/layer5io/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)

**Comment by leecalcote** [comment]
@hexxdump do you consider that this is an effort that you'll revisit (ever)?

Will you confirm your ability to champion the feature epic in Meshery v0.7.0 - https://discuss.meshery.io/t/meshery-v0-7-0-roadmap/232

**Comment by hexxdump** [comment]
> @hexxdump do you consider that this is an effort that you'll revisit (ever)?
> 
> Will you confirm your ability to champion the feature epic in Meshery v0.7.0 - https://discuss.meshery.io/t/meshery-v0-7-0-roadmap/232

Yes @leecalcote . I want to work on this. I will probably get to this in February, as I am bit busy this month.

**Comment by hexxdump** [comment]
> @hexxdump ready to begin this item?

yes @leecalcote

**Comment by Althaf66** [comment]
@leecalcote, I would like to work on this issue.

**Comment by leecalcote** [comment]
@hexxdump, what did your research produce in context of the new mechanism that AWS has recently provided for authentication?

**Comment by leecalcote** [comment]
@Althaf66, great. Please present a plan.

**Comment by Althaf66** [comment]
Hi @leecalcote,'aws eks update-kubeconfig --region --name -kubeconfig' command basically updates the kubeconfig file to use your cluster as the current context. My plan is to
- Load the default AWS config file with specified region set
- Create an Amazon EKS client
- updating the cluster config in the .meshery/kubeconfig file

**Comment by ishwar170695** [comment]
I reviewed the current implementation to check whether this issue is still applicable.

The original `mesheryctl system config eks` command is now deprecated. However, the underlying dependency on the AWS CLI still appears in the active EKS connection flow.

In particular, `mesheryctl connection create --type eks` still invokes the AWS CLI to configure the EKS kubeconfig rather than using the AWS Go SDK directly.

The original issue therefore appears to remain relevant, but its scope should be updated to target the current EKS connection creation flow rather than the deprecated `system config eks` command.

The remaining work appears to be replacing the AWS CLI invocation with the AWS Go SDK/client-go implementation in the active EKS connection flow.

**Comment by KDeekshita** [comment]
hi @hexxdump if its still open can u assign me i want to contribute!
